src/librustc_mir/transform/copy_prop.rs

```diff
@@ -175,6 +181,14 @@ impl<'tcx> MirPass<'tcx> for CopyPropagation {
                 break
             }
         }
+
+        // Strip out nops
+        for blk in mir.basic_blocks_mut() {
+            blk.statements.retain(|stmt| if let StatementKind::Nop = stmt.kind {
+                false
+            } else {
+                true
+            })
```

Comment:
Maybe we should just have an `is_nop()` function on `Statement`. I probably should have written one, sorry about that.